File: CPP/TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
	vector <int> result;
        for(int i=0;i<nums.size();i++)
        {
           for(int j=i;j<nums.size();j++)
           {
               if(i==j)
                   continue;

               if((nums[i]+nums[j])==target)
               {
                   result.push_back(i);
                   result.push_back(j);
               }
           }
        }
        return result;
    }
};

File: Rust/two-sum.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
    let nums_len = nums.len();
    let mut map = HashMap::new();

    for i in 0..nums_len {
        let t = target - nums[i];

        if let Some(x) = map.get(&t) {
            return vec![*x, i as i32];
        } else {
            map.insert(nums[i], i as i32);
        }
    }
    vec![]
}
